x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/utils/SimpleBoundedInputStream.java

```diff
@@ -38,6 +39,9 @@ public SimpleBoundedInputStream(InputStream inputStream, long maxBytes) {
     public int read() throws IOException {
         // We have reached the maximum, signal stream completion.
         if (numBytes >= maxBytes) {
+            if (throwWhenExceeded) {
+                throw new IOException("input stream exceeded maximum bytes of [" + maxBytes +"]");
+            }
             return -1;
```

Comment:
Why do we sometimes throw and sometimes return `-1`? Would it be possible to have only one exit point?